models/jira__issue_status_transitions.sql

```diff
@@ -0,0 +1,111 @@
+  where previous_status is null or previous_status != status
+),
+
+-- Add sequence and transition tracking
+status_transitions as (
+
+  select
+    issue_id,
+    status,
+    status_category_name,
+    valid_from as transition_at,
+    valid_until as transition_until,
+    is_current_status,
+    seconds_in_status,
+
+    -- Sequence tracking
+    row_number() over (partition by issue_id order by valid_from) as status_sequence,
+
+    -- Previous status tracking
+    previous_status,
+    lag(status_category_name) over (partition by issue_id order by valid_from) as previous_status_category_name,
+    lag(valid_from) over (partition by issue_id order by valid_from) as previous_transition_at,
+
+    -- Time in previous status
+    lag(seconds_in_status) over (partition by issue_id order by valid_from) as seconds_in_previous_status
+
+  from issue_status_history
+),
+
+-- Add transition type indicators and time calculations
+final as (
+
+  select
+    issue_id,
+    status,
+    status_category_name,
+    previous_status,
+    previous_status_category_name,
+    transition_at,
+    transition_until,
+    status_sequence,
+    is_current_status, 
+
+    -- Time spent in current status
+    round(cast(seconds_in_status as {{ dbt.type_numeric() }}) / 60, 2) as minutes_in_status,
+
+    -- Time spent in previous status (when transitioning)
+    case when seconds_in_previous_status is not null
+      then round(cast(seconds_in_previous_status as {{ dbt.type_numeric() }}) / 60, 2)
+      else null end as minutes_in_previous_status,
+
+    -- Workflow direction indicators
+    case
+      when previous_status_category_name is null then 'new'
+      when previous_status_category_name = 'To Do' and status_category_name = 'In Progress' then 'forward'
+      when previous_status_category_name = 'In Progress' and status_category_name = 'Done' then 'forward'
+      when previous_status_category_name = 'To Do' and status_category_name = 'Done' then 'forward'
+      when previous_status_category_name = 'Done' and status_category_name = 'In Progress' then 'backward'
+      when previous_status_category_name = 'In Progress' and status_category_name = 'To Do' then 'backward'
+      when previous_status_category_name = 'Done' and status_category_name = 'To Do' then 'backward'
+      when previous_status_category_name = status_category_name then 'lateral'
+      else 'other'
+    end as transition_direction,
+
+    -- Key lifecycle transitions
+    case when previous_status_category_name is null and status_category_name = 'To Do' then 1 else 0 end as added_work,
+    case when previous_status_category_name != 'In Progress' and status_category_name = 'In Progress' then 1 else 0 end as started_work,
+    case when previous_status_category_name != 'Done' and status_category_name = 'Done' then 1 else 0 end as completed_work,
+    case when previous_status_category_name = 'Done' and status_category_name != 'Done' then 1 else 0 end as reopened_work
```

Comment:
I'm not sure these add that much value from what's already provided. This creates a wider end model and can be confusing on how to exactly interpret these metrics. I would recommend removing these unless there's a strong case to keep them.

Reply:
Yeah I was trying to get more directional metrics, but I can remove these if we consider them superfluous and a net-neutral at best. @fivetran-savage is actually using these dimensions in her analysis, so maybe I'll ask if she wants to include those metrics.